frontend/src/pages/NewRunV2.tsx

```diff
@@ -311,14 +248,14 @@ function NewRunV2(props: NewRunV2Props) {
       <div className={commonCss.scrollContainer}>
         <div className={commonCss.header}>Run details</div>
 
-        {!pipelineId && apiResourceRefFromRun && !apiResourceRefFromRun[1] && (
+        {!apiPipeline && (
```

Comment:
It is probably better to check whether user is coming from run detail page or pipeline detail page, so we are confident to show `{usePipelineFromRunLabel}` if they are coming from run detail page.